code/headers/test_usart.cpp

```diff
@@ -0,0 +1,41 @@
+#include "test_usart.hpp"
+
+namespace r2d2 {
+
+    test_usart_c::test_usart_c(unsigned int baudrate, usart_ports_c controller):
+        baudrate(baudrate),
+        controller(controller)
+    {}
+
+    void test_usart_c::enable(){
```

Comment:
if you define the functions in the base class without the =0 you dont need to overide the functions with empty body not necessary to change for now but might be usefull in the future

Reply:
welp, I removed these overrides because I got an undefined vtable error, will add them

Reply:
this is better but did not mean this i was not very clear sorry, i meant that if you dont make them =0 in the base class you can basicly skip this in de inherited classes if you dont need the functions but this solves my problem for now.